### ISSUE TITLE
Show the correct example in SwaggerUI when using ash_money

### ISSUE THREAD

**Issue by aifrak** [ash-project/ash_money#69]
**Is your feature request related to a problem? Please describe.**
I have noticed that the example generated by SwaggerUI for the money type of `ash_money` is `string`. 

So currently, the generated example gives me this:
```json
"price": "string"
````

But the actual response for money type looks like this:
```json
"price": {
  "currency": "THB",
  "amount": "50.92"
}
```

**Describe the solution you'd like**
So in my mind, the generated example should have looked like this:

```json
"price": {
  "currency": "string",
  "amount": "string"
}
```

I have made a branch in a public repository [here](https://github.com/aifrak/phoenix-ash-library/tree/ash-json-api-with-ash-money) with this case.

**Describe alternatives you've considered**
I am not sure here, but if `ash_money` does not automatically give us the right example money type, maybe we should have somehow the possibility to define ourselves the money type.

**Express the feature either with a change to resource syntax, or with a change to the resource interface**
It is just an idea here as alternative way, but if we need to define the money type example, maybe we should add some callback like for a custom type.

```elixir
defmodule MyApp.Types.Money do
  use Ash.Type.Money

  @impl true
  def json_api_type_example() do # I am not sure about the name
    %{
	  "currency": "string",
	  "amount": "string"
    }
  end
end
```

**Additional context**
None

**Comment by zachdaniel** [comment]
Hey @aifrak if you update `AshJsonApi` and `AshMoney`, the type is now properly displayed in the open api schema :)

**Comment by aifrak** [comment]
Yes, I confirm it is working! You are so fast! Thank you so much!

**Comment by aifrak** [comment]
Sorry. I just have noticed a small detail. The example says amount is `0`, so an integer:
```jsonc
      "price": {
        "amount": 0, { # <-- Here
        "currency": "string"
      }
```

But the value I get is actually `string`:
```jsonc
     "price": {
        "amount": "50.92" # <-- Here
        "currency": "THB"
      }
```

Should I open another issue or is it something related to this one?

**Comment by zachdaniel** [comment]
Ah, you're totally right, thanks. I've fixed it in `main` of `ash_money`, but given that it comes with a `pattern`, it can add a bit of complexity and I want to check with some colleagues about how right that regex is. Feel free to try out `main` in the meantime.

**Comment by zachdaniel** [comment]
Fix released in `main` of `ash_money`

**Comment by aifrak** [comment]
I can confirm it is working! The example are even better than just `string`:
```json
      "price": {
        "amount": "100.00",
        "currency": "USD"
      }
```

Thank you very much again!
